network-api/networkapi/wagtailpages/pagemodels/customblocks/image_teaser_block.py

```diff
@@ -1,9 +1,17 @@
 from django.forms.utils import ErrorList
+from django.utils import functional as func_utils
+from django.utils import text as text_utils
 from wagtail.core import blocks
 from wagtail.core.blocks.struct_block import StructBlockValidationError
 from wagtail.images.blocks import ImageChooserBlock
 
 
+class ImageTeaserValue(blocks.StructValue):
+    @func_utils.cached_property
+    def slug(self):
+        return text_utils.slugify(self.get("title", ""))
+
+
```

Comment:
This is neat - I would have slugged the title and saved it as a slud value, but I prefer this a lot more!

Reply:
Yea, I thought it's unlikely for us to retrieve the blocks by slug, so a prop felt simpler.